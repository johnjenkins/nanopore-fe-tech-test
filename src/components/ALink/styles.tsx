import Link from 'next/link'
import styled from 'styled-components'

export const AlteredLink = styled(Link)`
  padding: 8px 10px;
  border-radius: 6px;
`
