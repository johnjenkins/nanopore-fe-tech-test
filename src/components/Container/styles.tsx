import styled from 'styled-components'

export const WrapperContainer = styled.div`
  margin: 0 auto;
  padding: 0 ${(props) => props.theme.spacing['16']};
  max-width: 100%;
  width: 100%;

  @media screen and (min-width: ${(props) => props.theme.breakpoints.xs}) {
    max-width: 540px;
  }

  @media screen and (min-width: ${(props) => props.theme.breakpoints.sm}) {
    max-width: 720px;
  }

  @media screen and (min-width: ${(props) => props.theme.breakpoints.md}) {
    max-width: 960px;
  }

  @media screen and (min-width: ${(props) => props.theme.breakpoints.lg}) {
    max-width: 1140px;
  }

  @media screen and (min-width: ${(props) => props.theme.breakpoints.xl}) {
    max-width: 1400px;
  }
`
