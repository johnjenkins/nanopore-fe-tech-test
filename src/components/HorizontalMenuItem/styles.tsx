import Link from 'next/link'
import styled from 'styled-components'

export const AlteredLink = styled(Link)`
  display: flex;
  align-items: center;
  justify-content: left;
  width: 100%;
  padding: 8px 10px;
  border-radius: 6px;
  &.active {
    background: #e4f1f5;
    font-weight: bold;
  }
`
