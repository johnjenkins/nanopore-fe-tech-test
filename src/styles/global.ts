import { createGlobalStyle } from 'styled-components'

export const theme = {
  colours: {
    primary: '#0c5a71',
    secondary: '#90c6e7',
    background: '#045a72'
  },
  breakpoints: {
    xs: '480px',
    sm: '768px',
    md: '992px',
    lg: '1200px',
    xl: '1440px'
  },
  spacing: {
    '2': '0.125rem',
    '4': '0.25rem',
    '14': '0.875rem',
    '16': '1rem',
    '18': '1.125rem',
    '20': '1.25rem',
    '21': '1.3125rem',
    '24': '1.5rem',
    '25': '1.5625rem',
    '30': '1.875rem',
    '32': '2rem',
    '250': '15.625rem',
    '300': '18.75rem',
    '320': '20rem',
    '720': '45rem'
  }
}

export const GlobalStyles = createGlobalStyle`
    *,
    *:before,
    *:after {
        margin: 0;
        padding: 0;
        outline: 0;
        box-sizing: border-box;
    }

    html,
    body,
    #__next {
        height: 100vh;
    }

    #__next {
        overflow-y: auto;
        font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
        background-color: ${theme.colours.background};
        color: white;
        width: 100%;
        min-height: 100%;
        text-align: center;

        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: ${theme.spacing['16']};
    }

    a,
    a:hover,
    a:visited {
        text-decoration: none;
        color: inherit;
    }

    a:hover {
        color: ${theme.colours.primary};
    }
`
