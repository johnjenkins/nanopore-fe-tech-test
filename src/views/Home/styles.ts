import styled from 'styled-components'

export const Logo = styled.img`
  width: ${(props) => props.theme.spacing['300']};
`

export const Title = styled.h1`
  color: ${(props) => props.theme.colours.secondary};
  font-size: ${(props) => props.theme.spacing['32']};
`

export const Description = styled.h2`
  font-size: ${(props) => props.theme.spacing['24']};
  line-height: 140%;
  font-weight: 400;
  width: min(${(props) => props.theme.spacing['720']}, 80%);
`
