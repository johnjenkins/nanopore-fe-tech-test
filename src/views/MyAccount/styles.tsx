import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import styled from 'styled-components'

export const LogoWrapper = styled.div`
  width: 200px;
  margin-bottom: 15px;
`
export const Logo = styled.img`
  width: 100%;
`

export const Header = styled.div`
  background: #005c76;
  display: flex;
`

export const Contents = styled.div`
  background: #ffffff;
  display: flex;
  padding: 15px;
  text-align: left;
  li {
    margin-top: 10px;
    margin-bottom: 10px;
  }
  a {
    font-size: ${(props) => props.theme.spacing['20']};
    color: #005c75;
  }
  hr {
    margin-top: 15px;
    margin-bottom: 15px;
  }
  h2 {
    color: black;
  }
`

export const LeftSidebar = styled.div`
  width: calc(20% - 15px);
  ul {
    list-style-type: none;
  }
`
export const Main = styled.div`
  margin-left: 20px;
  flex: 1;
`

export const UserWrapper = styled.div`
  padding: 10px;
  min-height: 80px;
  background: #ffffff;
  width: 20%;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  color: #1890b2;
  font-weight: bold;
  img {
    border-radius: 100%;
  }
  .userDetails {
    margin-left: 20px;
    display: flex;
    flex-direction: column;
    flex: 1;
    text-align: left;
    div:nth-child(2) {
      font-weight: 300;
      font-size: 14px;
      margin-top: 5px;
    }
  }
  select {
    width: 100%;
    font-size: 16px;
    background-color: #fff;
    border: 1px solid #89c5d7;
    padding: 5px;
    border-radius: 7px;
    margin-top: 10px;
  }
`

export const MenuWrapper = styled.div`
  background: #045a72;
  color: #7cb8d8;
  font-weight: bold;
  flex: 1;
  text-align: left;
  padding-left: 40px;
  h2 {
    color: white;
    font-size: ${(props) => props.theme.spacing['20']};
  }
`

export const UserDetails = styled.div`
  display: flex;
`

export const Navigation = styled.div`
  margin-top: 30px;
`

export const StyledFaIcon = styled(FontAwesomeIcon)`
  margin-right: 20px;
`

export const Table = styled.table`
  border-collapse: collapse;
  width: 100%;
  margin: 0 auto;
  color: black;

  th {
    background-color: #e4f1f5;
    text-align: left;
    padding: 8px;
    border-bottom: 1px solid #ddd;
  }

  td {
    padding: 8px;
    border-bottom: 1px solid #ddd;
  }

  tr:nth-child(even) {
    background-color: #f2f2f2;
  }

  tr:hover {
    background-color: #ddd;
  }
`

export const StyledButton = styled.button`
  background: transparent;
  border: 1px solid #89c5d7;
  padding: 7px;
  border-radius: 7px;
  &:hover {
    cursor: pointer;
  }
`

export const RoleDropdown = styled.select`
  background: transparent;
  border: 1px solid #89c5d7;
  padding: 7px;
  border-radius: 7px;
`
