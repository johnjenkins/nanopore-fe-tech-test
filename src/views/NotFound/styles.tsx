import Link from 'next/link'
import styled from 'styled-components'

export const Title = styled.h1`
  color: ${(props) => props.theme.colours.primary};
`

export const Subtitle = styled.p`
  font-size: ${(props) => props.theme.spacing['24']};
  margin: ${(props) => props.theme.spacing['20']} 0;
`

export const Home = styled(Link)`
  font-size: ${(props) => props.theme.spacing['24']};
`
