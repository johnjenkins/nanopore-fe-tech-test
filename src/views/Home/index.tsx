import ALink from 'components/ALink'
import Button from 'components/Button'
import Link from 'next/link'

import manifest from '../../../public/manifest.json'
import { Logo, Title, Description } from './styles'

const Home = ({ name = manifest.name, description = manifest.description }) => (
  <>
    <Link href="/">
      <Logo src="/img/ont-logo.svg" alt="Oxford nanopore Tech logo" />
    </Link>
    <Title>{name}</Title>
    <Description>{description}</Description>
    <ALink href="/my-account" passHref>
      <Button>Customers & Accounts</Button>
    </ALink>
  </>
)

export default Home
