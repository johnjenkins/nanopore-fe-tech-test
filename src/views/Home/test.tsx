import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import Home from '.'
import manifest from '../../../public/manifest.json'
import { theme } from '../../styles/global'

describe('<Home />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    )

    expect(
      screen.getByRole('heading', { name: manifest.name })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    )

    expect(container.childNodes[1]).toHaveStyleRule('color', '#90c6e7')
  })
})
