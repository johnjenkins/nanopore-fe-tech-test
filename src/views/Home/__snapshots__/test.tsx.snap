// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the heading 1`] = `
.c3 {
  padding: 8px 10px;
  border-radius: 6px;
}

.c4 {
  background-color: transparent;
  color: white;
  border: 0.125rem solid #90c6e7;
  border-radius: 0.25rem;
  padding: 1rem 1.5rem;
  font-size: 1.25rem;
  font-weight: 600;
  cursor: pointer;
  -webkit-transition: background-color 0.25s,color 0.5s;
  transition: background-color 0.25s,color 0.5s;
}

.c4:hover,
.c4:focus {
  background-color: #90c6e7;
  color: #045a72;
}

.c0 {
  width: 18.75rem;
}

.c1 {
  color: #90c6e7;
  font-size: 2rem;
}

.c2 {
  font-size: 1.5rem;
  line-height: 140%;
  font-weight: 400;
  width: min(45rem,80%);
}

<div>
  <a
    href="/"
  >
    <img
      alt="Oxford nanopore Tech logo"
      class="c0"
      src="/img/ont-logo.svg"
    />
  </a>
  <h1
    class="c1"
  >
    An Oxford Nanopore Frontend Technical Test
  </h1>
  <h2
    class="c2"
  >
    As part of our application process, we'd like you to demonstrate your talents with a small assignment. The task is useful for us to get an idea of your problem solving, organisation, eye-for-detail and overall thought processes. Please do not spend more than a few hours on it - we do not expect candidates to 'finish'; just get as far as you can.
  </h2>
  <a
    class="c3"
    href="/my-account"
  >
    <button
      class="c4"
    >
      Customers & Accounts
    </button>
  </a>
</div>
`;
